### PR TITLE
Move to editor IErrorTag API

### PR DESCRIPTION
**Pull request by rchande**
Looks like this API is available in P1. Instead of drawing our own suggestion adornment, we can simply return an IErrorTag with a well defined tag name and the editor does all the heavy lifting.

Tag @dotnet/roslyn-ide for review

### CODE REVIEW

**Comment by rchande** [comment]
@Pilchie Did we have a bug tracking the fact that quickinfo didn't work on suggestions?

**Comment by rchande** [comment]
QuickInfo works on suggestion dots now:
image

File: src/EditorFeatures/Core/Implementation/Diagnostics/DiagnosticsSuggestionTaggerProvider.cs [hunk 1]
```diff
@@ -44,36 +40,16 @@ internal partial class DiagnosticsSuggestionTaggerProvider :
...
-        {
-            lock (_suggestionTagGate)
-            {
-                _suggestionTag = new SuggestionTag(_editorFormatMap);
-            }
-        }
-
-        protected override ITaggerEventSource GetTaggerEventSource()
-        {
-            return TaggerEventSources.OnEditorFormatMapChanged(
```
Comment (CyrusNajmabadi) [inline src/EditorFeatures/Core/Implementation/Diagnostics/DiagnosticsSuggestionTaggerProvider.cs]: presumably we got rid of this event as well?

File: src/EditorFeatures/Core/Implementation/Diagnostics/DiagnosticsSuggestionTaggerProvider.cs [hunk 2]
```diff
@@ -44,36 +40,16 @@ internal partial class DiagnosticsSuggestionTaggerProvider :
...
         }
 
-        protected override SuggestionTag CreateTag(DiagnosticData diagnostic)
+        protected override IErrorTag CreateTag(DiagnosticData diagnostic)
         {
-            lock (_suggestionTagGate)
-            {
-                return _suggestionTag;
-            }
+            return new ErrorTag(PredefinedErrorTypeNames.HintedSuggestion, diagnostic.Message);
```
Comment (CyrusNajmabadi) [inline src/EditorFeatures/Core/Implementation/Diagnostics/DiagnosticsSuggestionTaggerProvider.cs]: nit: use =>

File: src/EditorFeatures/Core/Implementation/Diagnostics/SuggestionAdornmentManagerProvider.cs [hunk 3]
```diff
@@ -1,46 +0,0 @@
-// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
```
Comment (CyrusNajmabadi) [inline src/EditorFeatures/Core/Implementation/Diagnostics/SuggestionAdornmentManagerProvider.cs]: nom nom nom

**Comment by CyrusNajmabadi** [comment]
Nice!

**Comment by CyrusNajmabadi** [comment]
@rchande Are you able to change the colors to pink?

File: src/EditorFeatures/Core/Implementation/Diagnostics/DiagnosticsSuggestionTaggerProvider.cs (hunk 1 above)
Comment (rchande) [inline src/EditorFeatures/Core/Implementation/Diagnostics/DiagnosticsSuggestionTaggerProvider.cs]: It's still used by line separators.

Comment (CyrusNajmabadi) [inline src/EditorFeatures/Core/Implementation/Diagnostics/DiagnosticsSuggestionTaggerProvider.cs]: Ah ok.

**Comment by rchande** [comment]
@CyrusNajmabadi Yes, colors can be changed:
image

**Comment by rchande** [comment]
tagging @MattGertz 

**Customer scenario**

Customer hovers over suggestion dots to see what the suggestion is but QuickInfo does not appear.

**Bugs this fixes:**

https://github.com/dotnet/roslyn/issues/20057

**Workarounds, if any**

The user can click on the lightbulb and try to figure out which item in it is applicable to the suggestion dots.

**Risk**

Low. We're consuming an editor API that is itself an adaptation of our dots adornment.

**Performance impact**

Low/none. We do exactly the same analysis as before, only we return a different type of tag.

**Is this a regression from a previous update?**

No.

**Root cause analysis:**

Our implementation was a custom adornment that had no way of triggering QuickInfo when hovering on it.

**How was the bug found?**

dogfooding.

**Comment by MattGertz** [comment]
@rchande @Pilchie I'm confused -- did this never work?  Is this part of the planned remaining editor work that JoC approved?  (I've lost track.)

**Comment by rchande** [comment]
@MattGertz Right, this never worked. Here's a screenshot of P2, hovering on the dots. Observe that there's no message about the diagnostic:
image
I believe the plan was to take this change as a bug fix, since we are in fact fixing a bug.

**Comment by MattGertz** [comment]
Will take to JoC today for final confirmation.

**Comment by rchande** [comment]
Needs to be merged concurrently with https://github.com/dotnet/roslyn-internal/issues/1895

**Pull request merged**